crates/dslab-iaas/src/core/host_manager.rs

```diff
@@ -288,12 +307,21 @@ impl HostManager {
                 memory_load: self.get_memory_load(self.ctx.time()),
                 recently_added_vms: mem::take(&mut self.recently_added_vms),
                 recently_removed_vms: mem::take(&mut self.recently_removed_vms),
-                recent_vm_status_changes: mem::take(&mut self.recent_vm_status_changes),
             },
             self.monitoring_id,
             self.sim_config.message_delay,
         );
-
+        for item in &self.recent_vm_status_changes {
```

Comment:
```suggestion
        for (vm_id, (status, _)) in self.recent_vm_status_changes.drain() {
```
Используйте мощь pattern matching и drain(), и тогда не нужно ниже ковыряться в item и очищать после цикла changes.